sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClient.java

```diff
@@ -337,6 +338,10 @@ private PublicClientApplication getPublicClientApplication(boolean sharedTokenCa
                 }
             } else if (authType.equalsIgnoreCase("DC")) {
 
+                if (isADFSTenant()) {
+                    return Mono.error(new CredentialUnavailableException("IntelliJCredential  "
+                                         + "authentication unavailable. ADFS tenant/authorities are not supported."));
```

Comment:
Should this error also apply to service principal authentication as well? I'm not familiar enough with the IntelliJ plug-in, but can you authenticate a service principal in an ADFS instance? Does it not try to actually authenticate the SP, but rather just store the credentials?

Reply:
So, the IDE Azure Plugin auth won't succeed as the IDE Plugin passes in invalid resources in the request for the Az Stack.  But, the service principal details will get stored in the filesystem and if used via the IntelliJCredential with correct scopes specified, it works fine.

Reply:
It's an interesting definition of "Works Fine", but I suppose we don't need to artificially guard against it.

Reply:
Yeah, since Service Principal auth is supported against Az Stack from the credential, we don't need to guard against it.
It'll be a rare scenario for the user to take this route.